sdk/iOS/src/MSQueuePushOperation.m

```diff
@@ -225,6 +225,43 @@ - (void) finishTableOperation:(MSTableOperation *)operation item:(NSDictionary *
                                               internalError:storeError];
                 }
             }
+            else {
+                // Operation was successful, but another operation is pending.
```

Comment:
I think you need to compare operation.item with item (minus system properties) for this to be ok to do, as we don't want to stomp on other server changes.